sql/core/src/main/scala/org/apache/spark/sql/execution/ExpandExec.scala

```diff
@@ -89,6 +89,8 @@ case class ExpandExec(
     child.asInstanceOf[CodegenSupport].inputRDDs()
   }
 
+  override protected def doConsumeInChainOfFunc: Boolean = false
```

Comment:
I might be not 100% sure about your intention though, I feel this is a little confusing because `ExpandExec` consume functions can be chained in gen'd code, right?

Reply:
The `doConsume` produces something like:

       |for (int $i = 0; $i < ${projections.length}; $i ++) {
       |  switch ($i) {
       |    ${cases.mkString("\n").trim}
       |  }
       |  $numOutput.add(1);
       |  ${consume(ctx, outputColumns)}
       |}

So the consume logic of its parent node is actually wrapped in a local for-loop. It has the same effect as not chain the next consume.

Reply:
yea, probably we might need to describe more about exceptional cases we can't use this optimization like `HashAggregateExec` in https://github.com/apache/spark/pull/18931/files#diff-28cb12941b992ff680c277c651b59aa0R204

Reply:
The good news is, the just merged #19324 simplifies the usage of `continue` in codegen. I'm now testing with it if I can remove this tricky part of `continue`.